### PR TITLE
feat: add system log

### PR DESCRIPTION
**Pull request by J1za**
## Linear Ticket
https://linear.app/autokitteh/issue/UI-747/system-log
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/deployments/sessions/table.tsx [hunk 1]
```diff
@@ -214,7 +215,7 @@ export const SessionsTable = () => {
 				</Frame>
 			</div>
 
-			<Button className={resizeClass} />
+			<button className={resizeClass} data-resize-id={resizeId} />
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/components/organisms/deployments/sessions/table.tsx:253]: Ok, I'll change that and create a component in a new PR

File: src/components/organisms/splitFrame.tsx [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+	const resizeHorizontalId = useId();
+	const resizeVerticalId = useId();
+	const [leftSideWidth] = useResize({
+		direction: "horizontal",
+		initial: 50,
+		max: 70,
+		min: 35,
+		id: resizeHorizontalId,
+	});
+	const [outputHeight] = useResize({ direction: "vertical", initial: 30, max: 90, min: 20, id: resizeVerticalId });
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/components/organisms/splitFrame.tsx:21]: added const for const [leftSideWidth] = useResize({
const [outputHeight] will be removed in next PR

File: src/components/templates/appLayout.tsx [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
 
 export const AppLayout = ({ className, topbarVariant }: { className?: string; topbarVariant?: TopbarType }) => {
-	const appLayoutClasses = cn("h-screen w-screen pr-5", className);
+	const appLayoutClasses = cn("h-screen w-screen pr-5 flex", className);
+	const resizeId = useId();
+	const [systemLogHeight] = useResize({
+		direction: "vertical",
+		initial: 25,
+		max: 100,
+		min: 15,
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/components/templates/appLayout.tsx]: added

File: src/components/organisms/outputTabs.tsx [hunk 4]
```diff
@@ -28,7 +28,7 @@ export const OutputTabs = () => {
 		<div className="flex h-full flex-col pb-3">
```
Comment (RonenMars) [inline src/components/organisms/outputTabs.tsx:28]: Please add
```suggestion
		{featureFlags.systemLog ? (<div className="flex h-full flex-col pb-3">
```

File: src/store/useLoggerStore.ts [hunk 5]
```diff
@@ -21,7 +21,12 @@ const store: StateCreator<LoggerStore> = (set) => ({
 		set(() => ({
 			logs: [],
 		})),
+	switchLogger: (enabled) =>
```
Comment (RonenMars) [inline src/store/useLoggerStore.ts:24]: maybe toggleLogger?
The switch is more like a change; here we're turning it on or off, which is a toggle.
Pick one of: `toggleLogger` or `showHideLogger` or `toggleLoggerVisibility`
Also, the param name should be `isEnabled` or `isVisible` or `newState` or `displayState`

File: src/components/organisms/outputTabs.tsx [hunk 6]
```diff
@@ -28,7 +28,7 @@ export const OutputTabs = () => {
...
 			<div
 				className={
-					`static top-4 flex h-8 select-none items-center gap-1 uppercase xl:gap-2 2xl:gap-4 3xl:gap-5 ` +
+					`static -my-2 flex h-14 select-none items-center gap-1 uppercase xl:gap-2 2xl:gap-4 3xl:gap-5 ` +
```
Comment (RonenMars) [inline src/components/organisms/outputTabs.tsx:31]: We should always avoid negative margins, is there any way we can make it without negative?

File: src/store/useLoggerStore.ts (hunk 5 above)
Comment (J1za) [reply src/store/useLoggerStore.ts:24]: Obviously, isLoggerEnabled is clearer, `isEnabled `is shorter, but it is not clear what it is. What do you think?

File: src/components/organisms/systemLog.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+		[LoggerLevel.info]: "",
+		[LoggerLevel.log]: "",
+		[LoggerLevel.warn]: "text-yellow-500",
+	} as const;
+
+	return (
+		<Frame className="h-full overflow-hidden border border-none px-7 py-4">
+			<div className="flex justify-between">
+				<Typography className="font-semibold" element="h2" size="large">
+					System Log
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/components/organisms/systemLog.tsx]: added

File: src/components/organisms/systemLog.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		[LoggerLevel.info]: "",
+		[LoggerLevel.log]: "",
+		[LoggerLevel.warn]: "text-yellow-500",
+	} as const;
+
+	return (
+		<Frame className="h-full overflow-hidden border border-none px-7 py-4">
+			<div className="flex justify-between">
+				<Typography className="font-semibold" element="h2" size="large">
+					{t("clear")}
```
Comment (RonenMars) [inline src/components/organisms/systemLog.tsx:29]: Should be "title", not "clear"

File: src/store/useLoggerStore.ts [hunk 9]
```diff
@@ -21,7 +21,12 @@ const store: StateCreator<LoggerStore> = (set) => ({
 		set(() => ({
 			logs: [],
 		})),
+	toggleLogger: (enabled) =>
+		set(() => ({
+			isLoggerEnabled: enabled,
+		})),
 	logs: [],
+	isLoggerEnabled: false,
```
Comment (RonenMars) [inline src/store/useLoggerStore.ts:29]:
```suggestion
	toggleLogger: (isLoggerEnabled) =>
		set(() => ({
			enabled: isLoggerEnabled,
		})),
	logs: [],
	enabled: false,
```

File: src/components/organisms/systemLog.tsx (hunk 8 above)
Comment (J1za) [reply src/components/organisms/systemLog.tsx]: changed

**Review by RonenMars — APPROVED** [review]

File: src/store/useLoggerStore.ts (hunk 9 above)
Comment (RonenMars) [reply src/store/useLoggerStore.ts:29]: agreed to keep that way

**Pull request merged**